pkg/process/metadata/workloadmeta/workloadmeta.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcessEntity is a placeholder workloadmeta.ProcessEntity.
+// It does not contain all the fields that the final entity will contain.
+type ProcessEntity struct {
+	pid      int32
+	cmdline  []string
+	language *languagemodels.Language
+}
+
+// WorkloadMetaExtractor handles enriching processes with
+type WorkloadMetaExtractor struct {
+	cache      map[string]*ProcessEntity
```

Comment:
Let's add a comment here to make it more explicit what we're using as cache key.

Reply:
Updated!